forem/forem
# Reconfigure session store, upgrade `redis-actionpack`

## What type of PR is this? (check all applicable)


## Description

Since December 2020 we have `redis-actionpack` and `redis-rack` pinned to specific versions in our `Gemfile` (comment updated recently):

```ruby
# NOTE: [@rhymes, @citizen428]: sign in doesn't work with redis-rack 2.1.0+ or with redis-actionpack 5.2.0+
# We need to investigate why exactly, in the meantime we are hardcoding these two dependencies
# to keep them at the latest working version
gem "redis-actionpack", "5.1.0" # Redis session store for ActionPack. Used for storing the Rails session in Redis.
gem "redis-rack", "2.0.6" # Redis Store for Rack applications
```

However, this prevents us from eventually upgrading to Rails 7 which requires a newer version of `redis-actionpack`.

I spent a lot of time on this over the past few days and now luckily stumbled upon [this comment](https://github.com/redis-store/redis-store/issues/342#issuecomment-780915357):

> Think there might be a discrepancy with how Rails 6 `signed: true` works vs the whole point of storing stuff in a centralized session store. In short, you probably don't want to use a centralized store if you're going to be signing and encrypting sessions, because a.) you don't need to, and b.) I don't know if it's possible.

In a further comment the author also points out that they removed the `signed: true` option from the examples in https://github.com/redis-store/redis-actionpack/blob/master/README.md#usage.

So this PR does the following:

* Update `config/initializers/session_store.rb` to remove the `signed: true` option
* Update `redis-actionpack` to 5.2.0
* Update `redis-rack` to 2.1.3
* Reorganize the `Gemfile` a little bit

In my local tests everything seems to work as expected but I guess we still may want to deploy this to a canary first, cc @forem/sre.

## Related Tickets & Documents

#15825

## QA Instructions, Screenshots, Recordings

1. Check out this branch
2. Make sure your existing session still works
3. Log out
4. Ensure you can log back in again 

### UI accessibility concerns?

No UI changes

## Added/updated tests?

- [X] No, and this is why: nothing to test here

## [Forem core team only] How will this change be communicated?

- [X] This change does not need to be communicated, and this is why not: this should just be a config change
